### ISSUE TITLE
Driving Skill : NPE in Road Traffic advanced.gaml with simple track

### ISSUE THREAD

**Issue by benoitgaudou** [gama-platform/gama#2442]
### Steps to reproduce
1. Run the Road Traffic Advanced.gml model from the Plugin models library, with the use of the simple track, experiment2d experiment. But this occurs also with the Manhattan track.

2. After some step, I get a NPE:

```
java.lang.NullPointerException
	at simtools.gaml.extensions.traffic.AdvancedDrivingSkill.primComputePath(AdvancedDrivingSkill.java:698)
	at gaml.additions.GamlAdditions.lambda$80(GamlAdditions.java:116)
	at msi.gaml.compilation.GamaHelper.run(GamaHelper.java:42)
	at msi.gaml.statements.PrimitiveStatement.privateExecuteIn(PrimitiveStatement.java:91)
	at msi.gaml.statements.AbstractStatement.executeOn(AbstractStatement.java:41)
	at msi.gaml.statements.AbstractStatementSequence.executeOn(AbstractStatementSequence.java:43)
	at msi.gama.runtime.ExecutionScope.execute(ExecutionScope.java:381)
	at msi.gaml.expressions.PrimitiveOperator.value(PrimitiveOperator.java:75)
	at msi.gaml.statements.SetStatement.privateExecuteIn(SetStatement.java:138)
	at msi.gaml.statements.AbstractStatement.executeOn(AbstractStatement.java:41)
	at msi.gaml.statements.AbstractStatementSequence.privateExecuteIn(AbstractStatementSequence.java:57)
	at msi.gaml.architecture.reflex.ReflexStatement.privateExecuteIn(ReflexStatement.java:80)
	at msi.gaml.statements.AbstractStatement.executeOn(AbstractStatement.java:41)
	at msi.gaml.statements.AbstractStatementSequence.executeOn(AbstractStatementSequence.java:43)
	at msi.gaml.architecture.reflex.ReflexArchitecture.executeReflexes(ReflexArchitecture.java:100)
	at msi.gaml.architecture.reflex.ReflexArchitecture.executeOn(ReflexArchitecture.java:91)
	at msi.gama.runtime.ExecutionScope.execute(ExecutionScope.java:381)
	at msi.gama.metamodel.agent.AbstractAgent.doStep(AbstractAgent.java:285)
	at msi.gama.metamodel.agent.MinimalAgent.doStep(MinimalAgent.java:224)
	at msi.gama.metamodel.agent.AbstractAgent.step(AbstractAgent.java:255)
	at msi.gama.runtime.ExecutionScope.lambda$1(ExecutionScope.java:427)
	at msi.gama.runtime.ExecutionScope.runAndCatch(ExecutionScope.java:459)
	at msi.gama.runtime.ExecutionScope.pushRunAndCatch(ExecutionScope.java:449)
	at msi.gama.runtime.ExecutionScope.step(ExecutionScope.java:427)
	at msi.gama.runtime.concurrent.GamaExecutorService.doStep(GamaExecutorService.java:180)
	at msi.gama.runtime.concurrent.GamaExecutorService.step(GamaExecutorService.java:154)
	at msi.gama.metamodel.population.GamaPopulation.stepAgents(GamaPopulation.java:179)
	at msi.gama.metamodel.population.GamaPopulation.step(GamaPopulation.java:174)
	at msi.gama.runtime.ExecutionScope.lambda$2(ExecutionScope.java:428)
	at msi.gama.runtime.ExecutionScope.runAndCatch(ExecutionScope.java:459)
	at msi.gama.runtime.ExecutionScope.step(ExecutionScope.java:428)
	at msi.gama.metamodel.agent.GamlAgent.stepSubPopulations(GamlAgent.java:89)
	at msi.gama.metamodel.agent.AbstractAgent.doStep(AbstractAgent.java:285)
	at msi.gama.metamodel.agent.MinimalAgent.doStep(MinimalAgent.java:224)
	at msi.gama.metamodel.agent.AbstractAgent.step(AbstractAgent.java:255)
	at msi.gama.runtime.concurrent.SimulationRunner.lambda$0(SimulationRunner.java:30)
	at msi.gama.runtime.concurrent.SimulationRunner.lambda$1(SimulationRunner.java:54)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
```

### System and version 
GAMA Git MAcosx

**Comment by AlexisDrogoul** [comment]
I tried to test if it  could be a consequence of the fix made for #2270 (by reverting it), but it is not the case...

**Comment by ptaillandier** [comment]
It was due to the problem of agent comparison  - hashcode computation - (that impacts among other the closest_to operator).
It is normally fixed now.
